radar-gateway/src/main/kotlin/org/radarbase/gateway/resource/FileUploadResource.kt

```diff
@@ -0,0 +1,47 @@
+package org.radarbase.gateway.resource
+
+import jakarta.ws.rs.POST
+import jakarta.ws.rs.Path
+import jakarta.ws.rs.PathParam
+import jakarta.ws.rs.core.Context
+import jakarta.ws.rs.core.Response
+import org.glassfish.jersey.media.multipart.FormDataContentDisposition
+import org.glassfish.jersey.media.multipart.FormDataParam
+import org.radarbase.gateway.inject.ProcessFileUpload
+import org.radarbase.gateway.service.storage.StorageService
+import org.slf4j.LoggerFactory
+import org.radarbase.gateway.utils.Paths.FILE_PATH
+import org.radarbase.gateway.utils.Paths.PROJECT_PATH
+import org.radarbase.gateway.utils.Paths.TOPIC_PATH
+import org.radarbase.gateway.utils.Paths.UPLOAD_PATH
+import org.radarbase.gateway.utils.Paths.USER_PATH
+import java.io.InputStream
+import java.net.URI
+
+@Path("")
+class FileUploadResource(
+    @Context private val storageService: StorageService? = null,
+) {
+    @ProcessFileUpload
+    @POST
+    @Path("/$PROJECT_PATH/{projectId}/$USER_PATH/{subjectId}/$FILE_PATH/$TOPIC_PATH/{topicId}/$UPLOAD_PATH")
```

Comment:
I think this path made sense in context of the appserver, but for gateway, we can just include the path params in the POST request body. See other endpoints in gateway for inspiration